.github/workflows/e2e_libvirt.yaml

```diff
@@ -0,0 +1,186 @@
+# (C) Copyright Confidential Containers Contributors 2023.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Run libvirt e2e tests.
+name: (Callable) libvirt e2e tests
+
+on:
+  workflow_call:
+    inputs:
+      qcow2_artifact:
+        required: true
+        type: string
+      install_directory_artifact:
+        description: The archive name of the install directory
+        default: ''
+        required: false
+        type: string
+
+env:
+  CLOUD_PROVIDER: libvirt
+  GO_VERSION: "1.20.7"
+
+jobs:
+  test:
+    runs-on: az-ubuntu-2204
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Setup Golang version ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.qcow2_artifact }}
+          path: podvm
+
+      - name: Get the install directory
+        if: ${{ inputs.install_directory_artifact != '' }}
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.install_directory_artifact }}
+          path: install
+
+      - name: Install Libvirt
+        run: |
+          set -o pipefail
+          echo "::group::Install packages"
+          sudo apt-get update
+          sudo apt-get install -y qemu-kvm libvirt-daemon-system libvirt-dev
```

Comment:
It might not be relevant on the azure version of ubuntu, but just to note that I had to enable the universe repo to install `libvirt-dev` (which I don't remember needing last year?)

Reply:
hi @stevenhorsman , the runner is a Ubuntu 22.04 (`Canonical:0001-com-ubuntu-server-jammy:22_04-lts-gen2:latest` image) and it doesn't need to enable such as repo.